File: Templates/template.r
#!/usr/bin/env Rscript
name<-sub(".*/","",sub("--file=","",commandArgs()[grep("--file=", commandArgs())]))
ptm<-proc.time()[3];

### check optparse
chk.pkg<-function(pkg){
	pkgs<-rownames(installed.packages())
	for(p in pkg)
	{
		if(!(p %in% pkgs))
		{
			stop("Packge`", p, "' required\nPlease refer\n",
				paste("  https://cran.r-project.org/web/packages/", p,
				"/index.html", sep=""), "\nor (Bioconductor package)\n",
				paste("  https://bioconductor.org/packages/release/bioc/html/", p,
				"/index.html", sep=""), "\nto install it.\n");
		}
	}
}

ld.pkg<-function(pkg){
	for(p in pkg) suppressPackageStartupMessages(library(p, character.only=T))
}

chk.pkg("optparse")
ld.pkg("optparse")

option_list=list(
	make_option(c("-i", "--input"), type="character", default=NULL, metavar="file",
		help="input file"),
	make_option(c("-o", "--output"), type="character", default=NULL, metavar="file",
		help="output file"),
	make_option(c("-x", "--xlab"), type="character", default="group1", metavar="string",
		help="x-label of output picture [default=%default]"),
	make_option(c("-n", "--height"), type="integer", default=3, metavar="number",
		help="number of iterations [default=%default]"),
	make_option(c("--height"), type="double", default=5, metavar="double",
		help="height of plot [default=%default]"),
	make_option(c("-q", "--quiet"), action="store_true", type="logical", default=FALSE,
		help="show more information [default=%default]")
);

opt.parser<-OptionParser(
	option_list=option_list,
	add_help_option=T, prog=name,
	description="\nDescription of this script",
	epilogue="Contact AUTHOR for suggestions and bug reports.\n");
opt=parse_args(opt.parser);

#check required packages
pkg<-c("ggplot2")
chk.pkg(pkg)
ld.pkg(pkg)

if (opt$quiet) options(warn=-1);

# common functions
info<-function(...) {
	if (opt$quiet) invisible(NULL) else cat(file=stderr(), format(Sys.time(), '\033[2m[%H:%M:%S]\033[0m '), ..., '\n')
}

# https://stackoverflow.com/a/51156986
dhms<-function(t) {
	paste(t %/% (60*60*24),
	paste(formatC(t %/% (60*60) %% 24, width = 2, format = 'd', flag = '0'),
	formatC(t %/% 60 %% 60, width = 2, format = 'd', flag = '0'),
	formatC(t %% 60, width = 2, format = 'd', flag = '0'),sep = ':'));
}

if (is.null(opt$input) || is.null(opt$output)){
	print_help(opt.parser);
	stop("Required input/output not specified", call.=F);
}

# main


#invisible(dev.off())
info(paste0('Total runtime: ', dhms((proc.time()-ptm)[3])));

File: Templates/template.py
#!/usr/bin/env python

__author__ = "AUTHOR"
__version__ = "0.1.0"
__license__ = "MIT"

import sys
import argparse
from datetime import datetime
start=datetime.now()
version="0.1.0"
verbose=False

def main():
    global verbose
    parser = argparse.ArgumentParser(description="description of your script", epilog="Contact AUTHOR for more info")
    group = parser.add_mutually_exclusive_group()
    parser.add_argument("-i", "--in", type=str, default=None, help="input file")
    parser.add_argument("-o", "--out", type=str, default=None, help="output file")
    parser.add_argument("-n", "--num", type=int, default=None, help="some int value")
    parser.add_argument("-c", "--choice", type=int, choices=[0, 1, 2], help="choices to pick")
    group.add_argument("-q", "--quiet", action="store_true", default=False, help="suppress verbose")
    group.add_argument("--verbose", action="store_true", default=False, help="increase output verbosity")
    parser.add_argument("-v", "--version", action="store_true", default=False, help="show program version")
    args = parser.parse_args()
    if args.verbose:
        verbose=True
    if args.version:
        print("version: ", version)
    print(args.num)



if __name__ == '__main__':
    main()
    #print(__file__)
    if verbose:
        print("TAT: ", datetime.now()-start)
